components/push/README.md

```diff
@@ -26,3 +26,114 @@ failures, it's currently not checked in.
  * Sqlite > 3.24
  * SqlCipher > 3.4
 
+## API
+
+## Initialization
+
+Calls are handled by the `PushManager`, which provides a handle for future calls.
+
+example:
+```kotlin
+
+import mozilla.appservices.push.(PushManager, BridgeTypes)
+
+// The following are mock calls for fetching application level configuration options.
+// "SenderID" is the native OS push message application identifier. See Native
+// messaging documentation for details.
+val sender_id = SystemConfigurationOptions.get("SenderID")
+
+// The "bridge type" is the identifier for the native OS push message system.
+// (e.g. FCM for Google Firebase Cloud Messaging, ADM for Amazon Direct Messaging,
+// etc.)
+val bridge_type = BridgeTypes.FCM
+
+// The "registration_id" is the native OS push message user registration number.
+// Native push message registration usually happens at application start, and returns
+// an opaque user identifier string. See Native messaging documentation for details.
+val registration_id = NativeMessagingSystem.register(sender_id)
+
+val push_manager = PushManager(
+    sender_id,
+    bridge_type,
+    registration_id
+)
+
+// It is strongly encouraged that the connection is verified at least once a day.
+// This will ensure that the server and UA have matching information regarding
+// subscriptions. This call usually returns quickly, but may take longer if the
+// UA has a large number of subscriptions and things have fallen out of sync.
+
+for change in push_manager.verify_connection() {
+    // fetch the subscriber from storage using the change[0] and
+    // notify them with a `pushsubscriptionchange` message containing the new
+    // endpoint change[1]
+}
+
+```
+
+## New subscription
+
+Before messages can be delivered, a new subscription must be requested. The subscription info block contains all the information a remote subscription provider service will need to encrypt and transmit a message to this user agent.
+
+example:
+```kotlin
+
+// Each new request must have a unique "channel" identifier. This channel helps
+// later identify recipients and aid in routing. A ChannelID is a UUID4 value.
+// the "scope" is the ServiceWorkerRegistration scope. This will be used
+// later for push notification management.
+val channelID = GUID.randomUUID()
```

Comment:
recalling our previous conversation: we should have the server create these for us. We could allow null channelIDs to subscribe -- I think we should even go further and remove channelID argument entirely (at least on the kotlin side).

(and this is a separate issue entirely from this PR :) )

Reply:
Yeah, I go back and forth about this. I still kinda feel that the ID should be generated at the site where it's most significant. For UAID, that's on the server, but for ChannelID, it's more on the UA side. I know that the existing UA code does what you say, and has the server generate the channelID, which it then uses to later do whatever storage indexing.

Not sure that's the best way, though, but yeah, we can roll a fix later if need be.